llvm/lib/Target/EVM/EVMCalculateModuleSize.cpp

```diff
@@ -0,0 +1,137 @@
+//===----- EVMCalculateModuleSize.cpp - Calculate module size --*- C++ -*--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A utility set for determining the overall size of a module.
+//
+//===----------------------------------------------------------------------===//
+
+#include "EVMCalculateModuleSize.h"
+#include "EVMMachineFunctionInfo.h"
+#include "EVMSubtarget.h"
+#include "MCTargetDesc/EVMMCTargetDesc.h"
+#include "llvm/CodeGen/MachineModuleInfo.h"
+#include "llvm/CodeGen/TargetInstrInfo.h"
+#include "llvm/CodeGen/TargetSubtargetInfo.h"
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+
+// This is the copied from AsmPrinter::isBlockOnlyReachableByFallthrough.
+static bool isBlockOnlyReachableByFallthrough(const MachineBasicBlock *MBB) {
```

Comment:
Just to check: could't `analyzeBranch()` or `MachineBasicBlock::getFallThrough()` help here?